templates/common/_base/files/ofport-request.yaml

```diff
@@ -0,0 +1,37 @@
+mode: 0755
+path: "/etc/NetworkManager/dispatcher.d/pre-up.d/ofport-request.sh"
+contents:
+  inline: |
+    #!/bin/bash
+    # Set interface ofport_request to guarantee stable ofport numbers. This is important for flow matches.
+    # Otherwise, another ofport number is assigned to the interface on every restart of NetworkManager.
+    # If the same request ID exists for ports without ofport_request set such as patch ports then these
+    # interfaces will be moved by OVS to a different ID. We choose a high ofport_request number so that such
+    # conflicts are extremely unlikely to happen.
+    # In the case of duplicate assignments by this script, this will not fail but instead one of the interfaces
+    # will take the assigned value and the other one will take the value that was preassigned to it.
+    # We must only have a single physical, NM managed physical interface on the bridge, so this case should
+    # never happen.
+    set -eux -o pipefail
+    if [[ "{{ .NetworkType }}" != "OVNKubernetes" ]]; then
+        exit 0
+    fi
+    
+    INTERFACE_NAME=$1
+    OPERATION=$2
+    
+    CONNECTION_UUID=$(nmcli -t -f  device,type,uuid conn | awk -F ':' '{if($1=="'${INTERFACE_NAME}'" && $2!~/^ovs*/) print $NF}')
+    if [ "$CONNECTION_UUID" == "" ]; then
+        exit 0
+    fi
+    
+    OVS_SLAVE_TYPE=$(nmcli -t -f connection.slave-type conn show "$CONNECTION_UUID" | awk -F ':' '{print $NF}')
+    if [ "$OVS_SLAVE_TYPE" != "ovs-port" ]; then
+        exit 0
+    fi
+    
+    # This will yield ofport_requests 5xxxx where xxxx is always fix for the same interface name
+    OFPORT_REQUEST=$(echo "$INTERFACE_NAME" | md5sum | awk '{print "5" substr(strtonum("0x" $1),0,4)}')
```

Comment:
I don't have a strong opinion about this. This does seem a more robust way of generating the ofport id. But what about sticking to 1 which would be the natural id the port would have otherwise?